### PR TITLE
More fixes after colibri imx7 testing

### PR DESCRIPTION
**Pull request by kacf**
I think this should be the last fixes. After this, I only have device specific settings left in my `local.conf`:

* `MENDER_MTDIDS = "nand0=gpmi-nand"`
* `MENDER_MTDPARTS = "gpmi-nand:512k(mx7-bcb),1536k(u-boot1)ro,1536k(u-boot2)ro,512k(u-boot-env),-(ubi)"`
* `MENDER_STORAGE_PEB_SIZE = "131072"`
* `MENDER_STORAGE_TOTAL_SIZE_MB = "512"`

I am not sure how strongly these are tied to the `colibri-imx7` machine type, especially the last one. We could potentially put these also into the `machine.conf`, but is it appropriate, or should the user set them?

No other changes are required to build for `colibri-imx7`.

See individual commits for details.

### CODE REVIEW

File: meta-mender-core/classes/mender-setup-image.inc [hunk 1]
```diff
@@ -4,8 +4,9 @@ IMAGE_FSTYPES_append = "${@bb.utils.contains('DISTRO_FEATURES', 'mender-image-ub
 IMAGE_FSTYPES_append = "${@bb.utils.contains('DISTRO_FEATURES', 'mender-image-uefi', ' uefiimg uefiimg.bmap', '', d)}"
 IMAGE_FSTYPES_append = "${@bb.utils.contains('DISTRO_FEATURES', 'mender-image-bios', ' biosimg biosimg.bmap', '', d)}"
 
-MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_x86 = " kernel-image"
-MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_x86-64 = " kernel-image"
+MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_x86 =    " kernel-image"
+MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_x86-64 = " kernel-image"
+MACHINE_ESSENTIAL_EXTRA_RDEPENDS_append_mender-image_arm =    " kernel-image kernel-devicetree"
```
Comment (mirzak) [inline meta-mender-core/classes/mender-setup-image.inc]: Probably need a `append_mender-image_aarch64` as well as that is used for 64-bit arm boards.

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/patches/0005-fw_env_main.c-Fix-incorrect-size-for-malloc-ed-strin.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
+From 17236152bf80436567bf3f1f27af3915364ec0b6 Mon Sep 17 00:00:00 2001
+From: Kristian Amlie <kristian.amlie@northern.tech>
+Date: Wed, 4 Apr 2018 10:01:04 +0200
+Subject: [PATCH 5/5] fw_env_main.c: Fix incorrect size for malloc'ed string.
+
+Using sizeof gives the size of the pointer only, not the string.
+
+Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>
```
Comment (mirzak) [inline meta-mender-toradex-nxp/recipes-bsp/u-boot/patches/0005-fw_env_main.c-Fix-incorrect-size-for-malloc-ed-strin.patch]: I know we have not really enforced this, but what do you think about adding `Upstream-status:` tags on these patches.

Based on https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines, Section "Patch Header Recommendations: Upstream-Status".

File: meta-mender-core/classes/mender-setup-image.inc (hunk 1 above)
Comment (kacf) [reply meta-mender-core/classes/mender-setup-image.inc]: Good point. I'm always thinking that `arm` covers it, but I believe you are right.

File: meta-mender-toradex-nxp/recipes-bsp/u-boot/patches/0005-fw_env_main.c-Fix-incorrect-size-for-malloc-ed-strin.patch (hunk 2 above)
Comment (kacf) [reply meta-mender-toradex-nxp/recipes-bsp/u-boot/patches/0005-fw_env_main.c-Fix-incorrect-size-for-malloc-ed-strin.patch]: Sure, I can add it for convenience. Our patch flow is too low that I would bother about enforcing this now though.

**Comment by kacf** [comment]
> We are in the process of removing the `colibri-mx7-mender` target right?

Yes, and I didn't use that target for my build, I only used `colibri-imx7`, and master branches from all sources. However, because of the EasyInstaller, the work of removing `colibri-imx7-mender` is happening on rocko, which doesn't have the auto-patcher for Flash, so I'm unsure how much overlap there is. I can ask Alf when I see him.

**Comment by kacf** [comment]
Updated!

**Comment by mirzak** [comment]
LGTM

**Pull request merged**